ravedude/src/config.rs

```diff
@@ -0,0 +1,181 @@
+    pub open_console: Option<bool>,
+    pub serial_baudrate: Option<NonZeroU32>,
+    pub port: Option<std::path::PathBuf>,
+    pub reset_delay: Option<u64>,
+    pub board: Option<String>,
+}
+
+impl RavedudeGeneralConfig {
+    pub fn apply_overrides(&mut self, args: &crate::Args) -> anyhow::Result<()> {
+        // command line args take priority over Ravedude.toml
+        if args.open_console {
+            self.open_console = Some(true);
+        }
+        if let Some(serial_baudrate) = args.baudrate {
+            self.serial_baudrate = Some(
+                NonZeroU32::new(serial_baudrate)
+                    .ok_or_else(|| anyhow::anyhow!("baudrate must not be 0"))?,
+            );
+        }
+        if let Some(port) = args.port.clone() {
+            self.port = Some(port);
+        }
+        if let Some(reset_delay) = args.reset_delay {
+            self.reset_delay = Some(reset_delay);
+        }
+        Ok(())
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub struct BoardConfig {
+    pub name: Option<String>,
+    pub inherit: Option<String>,
+    pub reset: Option<ResetOptions>,
+    pub avrdude: Option<BoardAvrdudeOptions>,
+    pub usb_info: Option<BoardUSBInfo>,
+}
+
+impl BoardConfig {
+    pub fn merge(self, base: BoardConfig) -> Self {
+        Self {
+            name: self.name.or(base.name),
+            // inherit is used to decide what BoardGeneralOptions to inherit and isn't used anywhere else
+            inherit: None,
+            reset: self.reset.or(base.reset),
+            avrdude: match self.avrdude {
+                Some(avrdude) => base.avrdude.map(|base_avrdude| avrdude.merge(base_avrdude)),
+                None => base.avrdude,
+            },
+            usb_info: self.usb_info.or(base.usb_info),
+        }
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct ResetOptions {
+    pub automatic: bool,
+    pub message: Option<String>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+#[serde(rename_all = "kebab-case")]
+pub struct BoardAvrdudeOptions {
+    pub programmer: Option<String>,
+    pub partno: Option<String>,
+    #[serde(
+        serialize_with = "serialize_baudrate",
+        deserialize_with = "deserialize_baudrate"
+    )]
+    pub baudrate: Option<Option<NonZeroU32>>,
+    pub do_chip_erase: Option<bool>,
+}
+impl BoardAvrdudeOptions {
+    pub fn merge(self, base: Self) -> Self {
+        Self {
+            programmer: self.programmer.or(base.programmer),
+            partno: self.partno.or(base.partno),
+            baudrate: self.baudrate.or(base.baudrate),
+            do_chip_erase: self.do_chip_erase.or(base.do_chip_erase),
+        }
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+#[serde(rename_all = "kebab-case")]
+pub enum BoardUSBInfo {
+    PortIds(Vec<BoardPortID>),
+    #[serde(rename = "error")]
+    Error(String),
```

Comment:
Should we really let the config specify the error message here?  I know the old ravedude had each board config manually error out, but we don't need to keep that without good reason.

I think it'd be better for the config to simply state that port guessing is not supported and ravedude then provides a standardized error message.

Reply:
> Should we really let the config specify the error message here?
Honestly, probably not. I just kept it in because that's how it was before and I didn't want to make modifications to the preexisting behavior.

I've removed it. Now the message is just
```
Warning: this board cannot reset itself.

Once reset, press ENTER here:
```
for all the boards without automatic reset. If the behavior should be changed, let me know!